### ISSUE TITLE
Navigate with relative routes

### ISSUE THREAD

**Issue by rocketraman** [hfhbd/routing-compose#132]
Please allow navigation with relative routes.

Routing deep in a tree should not require knowledge of the entire parent routing tree. Example:

```kt
@Composable
fun Root() {
Hashrouter(initRoute = "/") {
  route("/") {
    Child1()
  }
}

@Routing
@Composable
fun NavBuilder.Child1() {
  route("/foo") {
    Child2()
  }
}


@Routing
@Composable
fun NavBuilder.Child2() {
  route("/bar") {
    A()
  }
  route("/baz") {
    B()
  }
}

@Routing
@Composable
fun NavBuilder.B() {
  val router = Router.current
  Button(attrs = { onClick { router.navigate("/foo/bar") } }) {
    Text("Go to A")
  }
}
```

Component `B` must now be aware of the routing tree all the way up to the root, including `/foo` and `Child1` which is a grand-parent node. However, with relative routing Component `B` just has to be aware that it is routing to a sibling:

```kt
@Routing
@Composable
fun NavBuilder.B() {
  val router = Router.current
  Button(attrs = { onClick { router.navigate("./bar") } }) {
    Text("Go to A")
  }
}
```

and if that routing decision is hoisted into `Child2`, it doesn't even need to be aware of that:

```kt
@Routing
@Composable
fun NavBuilder.Child2() {
  route("/bar") {
    A()
  }
  route("/baz") {
    B(toA = "./bar")
  }
}

@Routing
@Composable
fun NavBuilder.B(toA: String) {
  val router = Router.current
  Button(attrs = { onClick { router.navigate(toA) } }) {
    Text("Go to A")
  }
}
```

As a related note, the `route` builders are relative based on their position in the tree (despite them starting with a `/`), so it feels inconsistent to define routes relatively, but define navigation absolutely.

**Comment by hfhbd** [comment]
relates to #130

**Comment by hfhbd** [comment]
Will be fixed with #157. Using `/bar` results into absolut routing, `bar` uses relative routing with the current router.
